roachtest: add benchmark for INSPECT under admission control

spilchen opened the pull request:
Add a new roachtest to measure the impact of admission control on INSPECT operations running concurrently with foreground workloads.

As part of this, introduce a cluster setting knob to control admission control behavior for INSPECT operations, allowing them to be run with either BulkLowQoS (enabled) or Normal QoS (disabled).

The test performs three INSPECT runs:
1. A calibration run (with no workload) to dynamically determine test duration.
2. An INSPECT run with admission control enabled (BulkLowQoS) during a foreground read workload.
3. An INSPECT run with admission control disabled (Normal QoS) during the same workload.

Each run measures and reports throughput (rows/s/CPU), enabling us to quantify:
- The performance impact of admission control on INSPECT operations.
- The effectiveness of admission control in protecting foreground workload latency.

The workload duration is dynamically calculated to ensure it's long enough to encompass both INSPECT runs.

Informs: #154457
Epic: CRDB-30356

Release note: none

rafiss reviewed (approved):
really nice tests! i just had some small questions 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/155144#-:-ObFR4_vEn4wwhy7FW2Z:b-ir10dp)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @spilchen)

___
*[`pkg/cmd/roachtest/tests/admission_control_inspect.go` line 45 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/155144#-ObFRX7XDK-dxxVy-l7f:-ObFRX7XDK-dxxVy-l7g:b-2bl985) ([raw file](https://github.com/cockroachdb/cockroach/blob/0f0344e49acca117288a2b08e5eadb560235e218/pkg/cmd/roachtest/tests/admission_control_inspect.go#L45)):*
> ```Go
> 
> 	return registry.TestSpec{
> 		Name:             fmt.Sprintf("inspect/admission-control/rows=%d", numRows),
> ```

nit: it may be useful to include the cpu count and node count in the name. for example, see the naming convention of other test names at https://roachperf.crdb.dev/ (e.g. `cdc/scan/catchup-cold/nodes=5/cpu=16/rows=1G/ranges=100/protocol=mux/format=json/sink=null`). it's helpful since we also need code in roachperf to display these charts, and the charting logic is based on the test name.
___
*[`pkg/cmd/roachtest/tests/admission_control_inspect.go` line 232 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/155144#-ObFQbtL2s1tcer6CI1X:-ObFQbtL2s1tcer6CI1Y:ba6yxvn) ([raw file](https://github.com/cockroachdb/cockroach/blob/9e5799671ade6758f0ecbbdd0fc9684bdd94b1cb/pkg/cmd/roachtest/tests/admission_control_inspect.go#L232)):*
> ```Go
> 			t.L().Printf("Setting workload duration to %v (based on calibration: %v)\n", workloadDuration, calibrationDuration)
> 
> 			// Start read-only foreground workload using kv (since bulkingest doesn't support reads).
> ```

is there a specific reason that we want a read only workload here? what would the issue be with having a workload with writes? perhaps the answer could be included in this code comment.
___
*[`pkg/cmd/roachtest/tests/admission_control_inspect.go` line 248 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/155144#-ObFQmiEDMFalrps7o_t:-ObFQmiEDMFalrps7o_u:b-lmx930) ([raw file](https://github.com/cockroachdb/cockroach/blob/9e5799671ade6758f0ecbbdd0fc9684bdd94b1cb/pkg/cmd/roachtest/tests/admission_control_inspect.go#L248)):*
> ```Go
> 			time.Sleep(baselineDuration)
> 
> 			_ = runInspect(
> ```

nit: for these calls to runInspect, it could be useful to keep the return value and log the duration. seeing the durtaion in logs might be useful when manually looking at logs.


<!-- Sent from Reviewable.io -->

spilchen committed:
roachtest: add benchmark for INSPECT under admission control

Add a new roachtest to measure the impact of admission control on
INSPECT operations running concurrently with foreground workloads.

As part of this, introduce a cluster setting knob to control admission
control behavior for INSPECT operations, allowing them to be run with
either BulkLowQoS (enabled) or Normal QoS (disabled).

The test performs three INSPECT runs:
1. A calibration run (with no workload) to dynamically determine test duration.
2. An INSPECT run with admission control enabled (BulkLowQoS) during a
   foreground read workload.
3. An INSPECT run with admission control disabled (Normal QoS) during the
   same workload.

Each run measures and reports throughput (rows/s/CPU), enabling us to
quantify:
- The performance impact of admission control on INSPECT operations.
- The effectiveness of admission control in protecting foreground workload
  latency.

The workload duration is dynamically calculated to ensure it's long enough
to encompass both INSPECT runs.

Informs: #154457
Epic: CRDB-30356

Release note: none

spilchen reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/155144)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @rafiss)

___
*[`pkg/cmd/roachtest/tests/admission_control_inspect.go` line 45 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/155144#-ObFRX7XDK-dxxVy-l7f:-ObXV89x4eA7CB4HHpdz:b54x2kk) ([raw file](https://github.com/cockroachdb/cockroach/blob/0f0344e49acca117288a2b08e5eadb560235e218/pkg/cmd/roachtest/tests/admission_control_inspect.go#L45)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

nit: it may be useful to include the cpu count and node count in the name. for example, see the naming convention of other test names at https://roachperf.crdb.dev/ (e.g. `cdc/scan/catchup-cold/nodes=5/cpu=16/rows=1G/ranges=100/protocol=mux/format=json/sink=null`). it's helpful since we also need code in roachperf to display these charts, and the charting logic is based on the test name.
</blockquote></details>

I'll use this naming convention for the other inspect roachtest too.
___
*[`pkg/cmd/roachtest/tests/admission_control_inspect.go` line 232 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/155144#-ObFQbtL2s1tcer6CI1X:-ObXVIa6-szRSe8GTJOe:b-otyddx) ([raw file](https://github.com/cockroachdb/cockroach/blob/9e5799671ade6758f0ecbbdd0fc9684bdd94b1cb/pkg/cmd/roachtest/tests/admission_control_inspect.go#L232)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

is there a specific reason that we want a read only workload here? what would the issue be with having a workload with writes? perhaps the answer could be included in this code comment.
</blockquote></details>

INSPECT is read-only, so I would expect contention with another workload to be only on reads too. Also, I wanted a static dataset since we run INSPECT multiple times. I wanted it doing the same amount of work each time, just under different load characteristics.
___
*[`pkg/cmd/roachtest/tests/admission_control_inspect.go` line 248 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/155144#-ObFQmiEDMFalrps7o_t:-ObXW3Cr0YvWvp0_MDmN:b-wkjhov) ([raw file](https://github.com/cockroachdb/cockroach/blob/9e5799671ade6758f0ecbbdd0fc9684bdd94b1cb/pkg/cmd/roachtest/tests/admission_control_inspect.go#L248)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

nit: for these calls to runInspect, it could be useful to keep the return value and log the duration. seeing the durtaion in logs might be useful when manually looking at logs.
</blockquote></details>

The `runInspect` captures that with this log message:
https://github.com/cockroachdb/cockroach/pull/155144/files#diff-094b401c27f4a7015bb40ca9c4c9f24b3d9708948251fc5a5b796e5c10d061b6R175-R179


<!-- Sent from Reviewable.io -->
